### DISCUSSION TITLE
Add cache plugin.

### DISCUSSION THREAD

**Discussion by cathaysia** [woodpecker-ci/woodpecker#3799] [Request a Feature] [answered]
I had readed woodpecker's document, it seems there has no a plugin provide cache operator. I checked the github action and found that the woodpeck's plugin lacks a post action. This means I have to restore and store manually. Are there plans to provide support for the cache plugin? In my environment, without cache, CI often fails due to network problems.

**Comment by qwerty287** [comment]
Could you please elaborate what you mean by "cache plugins"?

General support for caching is definitely planned, but nobody is working on this. Take a look at the other issues and discussions for more info.

**Reply by cathaysia** [reply]
I used github cache plugin much: https://github.com/marketplace/actions/cache

It can restore cache at the start of pipeline, then store cache at then end of pipeline.

**Reply by leganck** [reply]
https://github.com/meltwater/drone-cache

**Comment by zc-devs** [comment]
> it seems there has no a plugin provide cache operator

https://github.com/drone-plugins/drone-s3

https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/upload-artifacts/upload-artifacts-to-s3/#download-artifacts-from-s3

**Reply by cathaysia** [reply]
Are we talking about woodpecker? 

Using s3 artifact does simulate the behavior of cache plugin, but it is not very convenient to use. I need to manually delete the directory, download the artifact, unzip it, and then upload it.

**Reply by zc-devs** [reply, marked as answer]
> cache 

Many wants the cache. Each has own description of cache. Define the "cache".
Maybe you want to proxy maven repository via Nexus. In that case you don't need a plugin.
Or maybe you want "cache" docker images. I don't know.


> Woodpecker has been originally forked from Drone 0.8

> Are we talking about woodpecker?

Yes, I suppose.

> manually delete the directory

Which directory?
Why do you need delete directory in WP's workspace, if it is created each time pipeline runs?
Does "manually" applies to all verbs (delete, download, unzip, upload) or only to "delete" part?

> (manually) download the artifact

Seems, not manually: https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/upload-artifacts/upload-artifacts-to-s3/#download-artifacts-from-s3

> unzip it

Yes, if you zipped it at upload. Or untar, or un-7z.
But plugin support to upload/download directory, as I understand.

> upload it

That exactly the plugin's responsibility...

---

That said, you can request downloading functionality from https://github.com/woodpecker-ci/plugin-s3, or archiving from https://github.com/drone-plugins/drone-s3.
BTW, should S3 plugin do archiving/compressing?
